src/main.py

```diff
@@ -34,8 +34,11 @@
     )
     # create the empty json file for images
     build_context = os.getenv("BUILD_CONTEXT")
+    ei_dedicated = os.getenv("EIA_DEDICATED") == "True"
     utils.write_to_json_file(constants.TEST_TYPE_IMAGES_PATH, {})
-    if args.framework not in build_config.DISABLE_FRAMEWORK_TESTS or build_context != "PR":
+    if (not ei_dedicated and args.framework not in build_config.DISABLE_FRAMEWORK_TESTS) or \
+       (ei_dedicated and args.framework not in build_config.DISABLE_EI_FRAMEWORK_TESTS) or \
```

Comment:
If there are any ei_dedicated entries: can we auto_populate DISABLE_FRAMEWORK_TESTS for non-EI containers? Until there's a more robust changeset-based solution, this should work. A warning message would also help that says: Build for one or more EI containers is enabled: Disabling all non-EI buid/test.

Reply:
Okay. I will only consider the scenario of either non-EI dedicated or EI dedicated.